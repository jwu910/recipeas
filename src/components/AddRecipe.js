import React, { Component } from 'react';

export default class AddRecipe extends Component {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleSubmit(event){
    event.preventDefault();
    console.log('submitted');
  }
  render() {
    const modalDisplay = {
      display: this.props.display ? 'block' : 'none'
    }
    return (
      <div
        id="myModal"
        style={modalDisplay}
        onClick={this.props.closeModal}>
        <form className="myModal-content container">
          <h4 className="myModal-header">Add a Recipe</h4>
          <div className="form-group">
            <label>Recipe</label>
            <input type="text" className="form-control" placeholder="Enter a name"/>
          </div>
          <div className="form-group">
            <label>Ingredients</label>
            <input type="text" className="form-control" placeholder="Enter ingredients, separated, by, commas"/>
          </div>
          <div className="pull-right">
            <button
              className="btn btn-primary"
              onClick={this.handleSubmit}>
              Add Recipe
            </button>
            <button
              className="btn btn-default close-modal"
              onClick={this.props.closeModal}>
              Close
            </button>
          </div>
        </form>
      </div>
    );
  }
}
