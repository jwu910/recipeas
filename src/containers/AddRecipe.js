import React, { Component } from 'react';
import { addRecipe } from '../actions';
import { connect } from 'react-redux';

export class AddRecipe extends Component {
  constructor(props) {
    super(props);

    this.state = {
      title: '',
      ingredients: ''
    }
    this.handleTitleChange = this.handleTitleChange.bind(this);
    this.handleIngredientsChange = this.handleIngredientsChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleTitleChange(event) {
    this.setState({ title: event.target.value });
  }
  handleIngredientsChange(event) {
    this.setState({ ingredients: event.target.value });
  }
  handleSubmit(event) {
    event.preventDefault();
    this.props.closeModal(event);
    this.props.dispatch(addRecipe(this.state));
  }
  render() {
    const modalDisplay = {
      display: this.props.display ? 'block' : 'none'
    }
    return (
      <div
        id="myModal"
        style={modalDisplay}
        onClick={this.props.closeModal}>
        <form className="myModal-content container">
          <h4 className="myModal-header">Add a Recipe</h4>
          <div className="form-group">
            <label>Recipe</label>
            <input
              type="text"
              className="form-control"
              placeholder="Enter a Title"
              value={this.state.title}
              onChange={this.handleTitleChange}
            />
          </div>
          <div className="form-group">
            <label>Ingredients</label>
            <input
              type="text"
              className="form-control"
              placeholder="Enter ingredients, separated, by, commas"
              value={this.state.ingredients}
              onChange={this.handleIngredientsChange}
            />
          </div>
          <div className="pull-right">
            <button
              className="btn btn-primary submit-recipe"
              onClick={this.handleSubmit}>
              Add Recipe
            </button>
            <button
              className="btn btn-default close-modal"
              onClick={this.props.closeModal}>
              Close
            </button>
          </div>
        </form>
      </div>
    );
  }
}

export default connect()(AddRecipe);
