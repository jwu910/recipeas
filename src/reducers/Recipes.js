import { ADD_RECIPE, REMOVE_RECIPE } from '../actions';

const defaultRecipes = [
  {
    title: 'Pumpkin Pie',
    ingredients: 'Pumpkin Puree, Sweetened Condensed Milk, Eggs, Pumpkin Pie Spice, Pie Crust'
  },
  {
    title: 'Spaghetti',
    ingredients: 'Noodles, Pasta Sauce, Meatballs'
  },
  {
    title: 'Peanut Butter Mug Cake',
    ingredients: 'Peanut Butter, Eggs, Baking Powder, Sugar'
  }
];

const Recipes = (state = defaultRecipes, action) => {
  switch (action.type) {
    case ADD_RECIPE:
      return [
        ...state,
        action.recipe
      ]
    case REMOVE_RECIPE:
      return state.filter(recipe => recipe.title !== action.recipeTitle);
    default:
      return state;
  }
}

export default Recipes;
