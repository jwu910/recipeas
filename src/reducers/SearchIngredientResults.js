import {
  ADD_SEARCH_INGREDIENT_RESULTS,
  CLEAR_SEARCH_INGREDIENT_RESULTS
} from '../actions/actionTypes';

const searchIngredientResults = (state = {}, action) => {
  switch (action.type) {
    case ADD_SEARCH_INGREDIENT_RESULTS:
      return Object.assign({}, state, action.results);
    case CLEAR_SEARCH_INGREDIENT_RESULTS:
      return {};
    default:
      return state;
  }
};

export default searchIngredientResults;
