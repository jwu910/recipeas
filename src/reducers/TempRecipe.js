import {
  MODIFY_TEMP_TITLE,
  ADD_TEMP_INGREDIENT,
  EDIT_TEMP_INGREDIENT,
  REMOVE_TEMP_INGREDIENT,
  CLEAR_TEMP_RECIPE
} from '../actions';

const initialTempRecipe = {
  titleDetails: {
    title: '',
    allergens: '',
    servings: ''
  },
  ingredients: []
};

const TempRecipe = (state = initialTempRecipe, action) => {
  switch(action.type) {
    // works for adding or editing
    case MODIFY_TEMP_TITLE:
      return Object.assign({}, state, action.titleDetails);
    case ADD_TEMP_INGREDIENT:
      return Object.assign({}, state, {
        ingredients: [
          ...state.ingredients,
          action.ingredient
        ]
      });
    case EDIT_TEMP_INGREDIENT:
      return Object.assign({}, state, {
        ingredients: state.ingredients.map(ingredient => {
          return ingredient.id === action.id ?
            { id: ingredient.id, ...action.ingredient }
            : ingredient;
        })
      });
    case REMOVE_TEMP_INGREDIENT:
      return Object.assign({}, state, {
        ingredients: state.ingredients.filter(ingredient => ingredient.id !== action.id)
      });
    case CLEAR_TEMP_RECIPE:
      return {};
    default:
      return state;
  }
};

export default TempRecipe;
