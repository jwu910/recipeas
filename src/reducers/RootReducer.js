import { combineReducers } from 'redux';
import Recipes from './Recipes';
import TempRecipe from './TempRecipe';
import SearchIngredientResults from './SearchIngredientResults';

const RootReducer = combineReducers({
  recipes: Recipes,
  tempRecipe: TempRecipe,
  results: SearchIngredientResults
});

export default RootReducer;
