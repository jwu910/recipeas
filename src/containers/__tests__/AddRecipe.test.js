import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { AddRecipe } from '../AddRecipe';
import configureStore from 'redux-mock-store';
import { defaultRecipes } from '../../reducers/Recipes';

const props = {
  display: true,
  closeModal: jest.fn(),
  dispatch: jest.fn()
};

const setup = (type) => {
  // setup only initializes the React component, not the redux container connected to store
  let wrapper;

  switch(type) {
    case 'shallow':
      wrapper = shallow(<AddRecipe {...props} />);
      break;
    case 'mount':
      wrapper = mount(<AddRecipe {...props} />);
      break;
    default:
      wrapper = undefined;
  }

  return {
    props,
    wrapper
  };
}

// calling onClick functions are not tested since it is in the scope of React's tests and the browser's test

describe('Add Recipe Menu', () => {
  test('Renders without crashing', () => {
    const component = renderer.create(<AddRecipe { ...props } />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Renders the connected AddRecipe component', () => {
    const store = configureStore()({
      recipes: defaultRecipes
    });

    const wrapper = mount(<AddRecipe { ...props } store={store} />);

    expect(wrapper.find(AddRecipe)).toHaveLength(1);
  });

  test('Displays invisibly when display props is false', () =>{
        const wrapper = shallow(<AddRecipe  { ...props } display={ false }/>);

        expect(wrapper.find('#myModal').node.props.style).toEqual({"display": "none"});
  });

  // testing that state updates properly //
  test('Updates title property in state correctly', () => {
    const { wrapper } = setup('shallow');
    const titleInputValue = {
      target: {
        value: 'BLT'
      }
    };

    const titleInput = wrapper.find('.form-control').at(0);

    titleInput.simulate('change', titleInputValue);

    expect(wrapper.state().title).toBe(titleInputValue.target.value);
  });

  test('Updates ingredients property in state correctly', () => {
    const { wrapper } = setup('shallow');
    const ingredientsInputValue = {
      target: {
        value: 'Bacon, Lettuce, Tomato'
      }
    }
    const ingredientsInput = wrapper.find('.form-control').at(1);

    ingredientsInput.simulate('change', ingredientsInputValue);

    expect(wrapper.state().ingredients).toBe(ingredientsInputValue.target.value);
  });

  // testing controlled components value
  test('Updates title value in input correctly', () => {
    const { wrapper } = setup('mount');
    const titleInputValue = {
      target: {
        value: 'BLT'
      }
    };
    const titleInput = wrapper.find('.form-control').at(0);
    titleInput.simulate('change', titleInputValue);

    expect(titleInput.props().value).toBe(titleInputValue.target.value);
  });

  test('Updates ingredients value in input correctly', () => {
    const { wrapper } = setup('mount');
    const ingredientsInputValue = {
      target: {
        value: 'Bacon, Lettuce, Tomato'
      }
    };
    const ingredientsInput = wrapper.find('.form-control').at(1);
    ingredientsInput.simulate('change', ingredientsInputValue);

    expect(ingredientsInput.props().value).toBe(ingredientsInputValue.target.value);
  });

});
