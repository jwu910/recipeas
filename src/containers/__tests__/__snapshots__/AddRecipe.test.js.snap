// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Recipe Menu Renders without crashing 1`] = `
<div
  id="myModal"
  onClick={[Function]}
  style={
    Object {
      "display": "block",
    }
  }
>
  <form
    className="myModal-content container"
  >
    <h4
      className="myModal-header"
    >
      Add a Recipe
    </h4>
    <div
      className="form-group"
    >
      <label>
        Recipe
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Enter a Title"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Ingredients
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Enter ingredients, separated, by, commas"
        type="text"
        value=""
      />
    </div>
    <div
      className="pull-right"
    >
      <button
        className="btn btn-primary submit-recipe"
        onClick={[Function]}
      >
        Add Recipe
      </button>
      <button
        className="btn btn-default close-modal"
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </form>
</div>
`;
